frontend/src/scenes/saved-insights/savedInsightsLogic.ts

```diff
@@ -12,18 +15,36 @@ interface InsightsResult {
 }
 
 export const savedInsightsLogic = kea<savedInsightsLogicType<InsightsResult>>({
+    actions: {
+        setInsightType: (type: string) => ({ type }),
+        setCreatedBy: (user: Partial<UserBasicType> | 'All users') => ({ user }),
+        setLayoutView: (view: string) => ({ view }),
+        setTab: (tab: string) => ({ tab }),
+        setDates: (dateFrom: string | Dayjs | undefined, dateTo: string | Dayjs | undefined) => ({
+            dateFrom,
+            dateTo,
+        }),
+        setSearchTerm: (term: string) => ({ term }),
+        renameInsight: (id: number) => ({ id }),
+        duplicateInsight: (insight: DashboardItemType) => ({ insight }),
+        addToDashboard: (item: DashboardItemType, dashboardId: number) => ({ item, dashboardId }),
+    },
     loaders: ({ values }) => ({
         insights: {
             __default: { results: [], count: 0 } as InsightsResult,
-            loadInsights: async (key?: string) => {
+            loadInsights: async () => {
                 const response = await api.get(
                     'api/insight/?' +
                         toParams({
                             order: '-created_at',
                             limit: 15,
                             saved: true,
-                            ...(key === SavedInsightsParamOptions.Yours && { user: true }),
-                            ...(key === SavedInsightsParamOptions.Favorites && { favorited: true }),
+                            ...(values.tab === SavedInsightsTabs.Yours && { user: true }),
+                            ...(values.tab === SavedInsightsTabs.Favorites && { favorited: true }),
+                            ...(values.searchTerm && { search: values.searchTerm }),
+                            ...(values.insightType.toLowerCase() !== 'all types' && { insight: values.insightType }),
+                            ...(values.createdBy !== 'All users' && { created_by: values.createdBy?.id }),
+                            ...(values.dates.dateFrom && { date_from: dayjs().subtract(2, 'day') }),
```

Comment:
This is just a static example but there has to be a better way of handling the time options than to translate each of them manually into dayjs dates?